packages/flutter/lib/src/widgets/animated_cross_fade.dart

```diff
@@ -189,49 +197,73 @@ class _AnimatedCrossFadeState extends State<AnimatedCrossFade> with TickerProvid
     }
   }
 
-  @override
-  Widget build(BuildContext context) {
-    List<Widget> children;
+  /// Whether we're in the middle of cross-fading right this frame.
+  bool get _isTransitioning => _controller.status == AnimationStatus.forward || _controller.status == AnimationStatus.reverse;
 
-    if (_controller.status == AnimationStatus.completed ||
-        _controller.status == AnimationStatus.forward) {
-      children = <Widget>[
-        new FadeTransition(
-          opacity: _secondAnimation,
-          child: widget.secondChild,
-        ),
-        new Positioned(
+  List<Widget> _buildCrossFadedChildren() {
+    const Key kFirstChildKey = const Key('first-child');
```

Comment:
don't use strings if you can help it (since keys are compared). A `ValueKey<CrossFadeState>` would probably do the job more efficiently. (At least, I hope that comparing an enum is faster than comparing a string...)